simplemodels/models.py

```diff
@@ -164,6 +165,36 @@ def _prepare_fields(self, kwargs):
         return kwargs
 
     @classmethod
+    def _get_path(cls, root, path, name):
+        """
+        Helper method to read value from nested data structure by given path.
+
+        :param root: root object with data
+        :param path: beginning of the path
+        :param name: last element name
+        :return: extracted value or None
+        """
+
+        if isinstance(path, six.string_types):
+            path = path.split('/')
+
+        path = list(filter(None, path)) if path else []
+        path = path + [name]
+        cur = root
+        for seg in path:
+            try:
+                cur = cur[seg]
+            except (KeyError, IndexError):
+                return None
+            except TypeError:
+                try:
```

Comment:
The only thing bothers me is the fact that if one would set `ALLOW_EXTRA_FIELDS` to `True`, then model would have duplicated data:

```python
#!/usr/bin/env python
# -*- coding: utf-8 -*-

from simplemodels.fields import CharField, IntegerField
from simplemodels.models import Document


class User(Document):
    class Meta(object):
        ALLOW_EXTRA_FIELDS = True

    name = CharField()
    salary = IntegerField(path='data/foo/bar/')


data = {
    'name': 'John Smith',
    'data': {
        'foo': {
            'bar': {
                'salary': 100500
            }
        }
    }
}

if __name__ == '__main__':
    user = User.create(data)
    print(user)
>>> {'data': {'foo': {'bar': {'salary': 100500}}}, 'name': 'John Smith', 'salary': 100500}
```

What do you think about it?